projeto/index.js

```diff
@@ -0,0 +1,105 @@
+console.log('--------------------------------------')
+console.log('     Projeto Carrinho de Compras     ')
+console.log('--------------------------------------')
+
+const readline = require('readline-sync')
+const db = require('./database')
+const { produtos } = require('./database')
+
+//Listar no console uma tabela contendo os produtos em ordem crescente de preço (do menor ao maior)
+produtos.sort(function (a, b) {
+    return a.preco - b.preco
+})
+console.table(produtos)
+console.log('--------------------------------------')
+
+const array = new Array() //irá lista o novo pedido
+
+//depois do NÂO ---- cria a classe pedido e trás as informações dentro da tabela  da compra.
+class Pedido {
+    constructor(array) {
+        this.listaProdutos = array
+        this.valorCupom = 0
+        this.dataPedido = new Date()
+    }
+    calcSubtotal() {
+        //Total da compra - //- Calcular o valor do subtotal (sem considerar o desconto)        
+        this.valorCupom = this.listaProdutos.reduce((acumulador, item) => acumulador + (item.preco * item.qtdeProduto), 0);
```

Comment:
`this. valorCupom = this.listaProdutos.reduce((acumulador, item) => acumulador + (item.preco * item.qtde), 0);`

O item veio como:
```
{
   id: 1,
   nome: 'Leite',
   descricao: 'Leite desnatado de 1 L',
   categoria: 'alimento',
   preco: 4,
   qtde: 1
}
```
Então, vc acessa o valor como `item.qtde` e não como `item.qtdeProduto`

Reply:
É verdade!!Obrigada!